pkg/controllers/cloud_config_sync_controller.go

```diff
@@ -60,39 +62,80 @@ func (r *CloudConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{}, nil
 	}
 
-	// Use kube-cloud-config from openshift-config-managed namespace as default source.
-	// If it is not exists try to use cloud-config reference from infra resource.
-	// https://github.com/openshift/library-go/blob/master/pkg/operator/configobserver/cloudprovider/observe_cloudprovider.go#L82
-	defaultSourceCMObjectKey := client.ObjectKey{
-		Name: managedCloudConfigMapName, Namespace: OpenshiftManagedConfigNamespace,
+	cloudConfigTransformerFn, err := cloud.GetCloudConfigTransformer(infra.Status.PlatformStatus)
+	if err != nil {
+		klog.Errorf("unable to get cloud config transformer function; unsupported platform")
+		if err := r.setDegradedCondition(ctx); err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to set conditions for cloud config controller: %v", err)
+		}
+		return ctrl.Result{}, err
 	}
-	sourceCM := &corev1.ConfigMap{}
-	if err := r.Get(ctx, defaultSourceCMObjectKey, sourceCM); errors.IsNotFound(err) {
-		klog.Warningf("managed cloud-config is not found, falling back to infrastructure config")
 
-		openshiftUnmanagedCMKey := client.ObjectKey{Name: infra.Spec.CloudConfig.Name, Namespace: OpenshiftConfigNamespace}
-		if err := r.Get(ctx, openshiftUnmanagedCMKey, sourceCM); err != nil {
-			klog.Errorf("unable to get cloud-config for sync")
+	sourceCM := &corev1.ConfigMap{}
+	managedConfigFound := false
+
+	// NOTE: We know that there is some transformation logic in place in CCO
```

Comment:
@stephenfin can you please specify what CCO means there. Leave repo link in this comment perhaps

Reply:
Whoops, this is [Cluster Config Operator](github.com/openshift/cluster-config-operator/). I can push a follow-up to clarify this if you'd like

Reply:
Fixed this